test/integration/sessions_test.rb

```diff
@@ -1,10 +1,22 @@
 require "integration_test_helper"
+require "gds_api/test_helpers/account_api"
 
 class SessionsTest < ActionDispatch::IntegrationTest
+  include GdsApi::TestHelpers::AccountApi
+
   context "when logged in" do
-    should "redirect the user to the account manager URL if they visit /sign-up" do
+    should "redirect the user to the account manager URL if they visit /sign-in" do
       get "/sign-in", headers: { "GOVUK-Account-Session" => "placeholder" }
       assert_redirected_to Plek.find("account-manager")
     end
   end
+
+  context "when logged out" do
+    should "add a redirect_path param to /sign-in from the HTTP REFERRER header" do
+      stub_account_api_get_sign_in_url(redirect_path: "/bank-holiday")
```

Comment:
I tried to think of a more explicit test here, perhaps by snooping on the `request`.
But in this case the test fails because webmock is unhappy with the path if the referer header and redirect path in the mock do not align.